src/main/java/isaac/bastion/listeners/ModeListener.java

```diff
@@ -0,0 +1,156 @@
+
+	private BottomLine bsiBottomLine;
+	private CivScoreBoard bsiBoard;
+	private BastionSettingManager settingMan;
+
+	public ModeListener() {
+		this.bsiBoard = ScoreBoardAPI.createBoard("bsiDisplay");
+		this.bsiBottomLine = BottomLineAPI.createBottomLine("bsiDisplay", 3);
+		this.settingMan = Bastion.getSettingManager();
+		settingMan.getBsiOverlay().registerListener(new SettingChangeListener<Boolean>() {
+			@Override
+			public void handle(UUID player, PlayerSetting<Boolean> setting, Boolean oldValue, Boolean newValue) {
+				if (newValue) {
+					checkLocationForBastions(Bukkit.getPlayer(player), Bukkit.getPlayer(player).getLocation());
+					return;
+				}
+				bsiBoard.hide(Bukkit.getPlayer(player));
+				bsiBottomLine.removePlayer(Bukkit.getPlayer(player));
+			}
+		});
+		settingMan.getBsiLocation().registerListener(new SettingChangeListener<String>() {
+			@Override
+			public void handle(UUID player, PlayerSetting<String> setting, String oldValue, String newValue) {
+				checkLocationForBastions(Bukkit.getPlayer(player), Bukkit.getPlayer(player).getLocation());
+			}
+		});
+	}
+
+	@EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
+	public void onPlayerJoin(PlayerJoinEvent pje) {
+		if (!settingMan.getBsiOverlay().getValue(pje.getPlayer())) {
+			return;
+		}
+		checkLocationForBastions(pje.getPlayer(), pje.getPlayer().getLocation());
+	}
+
+	@EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
+	public void onPlayerMove(PlayerMoveEvent pme) {
+		Location from = pme.getFrom();
+		Location to = pme.getTo();
+		if (to == null) {
+			return;
+		}
+
+		if (from.getBlockX() == to.getBlockX() && from.getBlockY() == to.getBlockY()
+				&& from.getBlockZ() == to.getBlockZ() && from.getWorld().equals(to.getWorld())) {
+			return;
+		}
+
+		if (!settingMan.getBsiOverlay().getValue(pme.getPlayer())) {
+			return;
+		}
+
+		checkLocationForBastions(pme.getPlayer(), to);
+	}
+
+	/**
+	 * This method handles the updating of the scoreboard/action bar
+	 * as to whether the player is in a bastion field or not.
+	 *
+	 * @param player to update bastion status to
+	 * @param location to check for bastions
+	 */
+	public void checkLocationForBastions(Player player, Location location) {
+		if (player == null || location == null) {
+			return;
+		}
+		Set<BastionBlock> bastionBlocks = Bastion.getBastionManager().getBlockingBastions(location);
+		if (bastionBlocks.isEmpty()) {
+			if (settingMan.getShowNoBastion(player.getUniqueId())) {
+				updateDisplaySetting(player, settingMan.getBsiLocation(), settingMan.getBsiOverlay().getValue(player.getUniqueId()),
+						ChatColor.WHITE + "No Bastion");
+				return;
+			}
+			bsiBoard.hide(player);
+			bsiBottomLine.removePlayer(player);
+			return;
+		}
+		boolean isFriendly = false;
+		boolean isEnemy = false;
+		for (BastionBlock bastions : bastionBlocks) {
+			if (!bastions.getType().isBlockReinforcements() || bastions.canPlace(player)) {
```

Comment:
This will show all claims bastions as enemy ones, you need to check whether the bastion blocks placement in general. Idk what the method name for that is, but it should look something like this
```
if (bastions.getType().blocksPlacing()) {
  //ignore any non-vault bastions
  continue;
}
if (bastions.canPlace(player) {
  isFriendly = true;
}
else {
  isEnemy = true;
}
```